integreat_cms/cms/utils/content_utils.py

```diff
@@ -0,0 +1,147 @@
+import logging
+from urllib.parse import urlparse
+
+from django.conf import settings
+from django.db.models import Q
+from lxml.html import HtmlElement, tostring
+
+from ..models import MediaFile
+from ..utils import internal_link_utils
+
+
+def clean_content(content: HtmlElement, language_slug: str) -> None:
```

Comment:
I would prefer if this function would not have side-effects and instead just take in the content as a string and return the cleaned content also as string. Since we don't really store  `HtmlElement` anywhere, every place that calls this function would have to convert the content to that first.

Reply:
I am not entirely sure I understand. What I proposed is that the signature of `clean_content` looks roughly like this:
```python
def clean_content(content: str, language_slug: str) -> str:
```
So there's no need to convert to an HtmlElement outside of this function

Reply:
I'm sorry for making it not understandable enough. I felt strange about changing it that way as in [these lines](https://github.com/digitalfabrik/integreat-cms/blob/develop/integreat_cms/cms/forms/custom_content_model_form.py#L111-L115) we just converted it from a string to an `HtmlElement` and then would need to change it back to a string right away. That made feel weird. Did I make my point clearer this time? :)

Reply:
There is no need to do that though. Just pass the string directly to `clean_content`.
For reference, here is what it looks like on the ical branch currently: https://github.com/digitalfabrik/integreat-cms/compare/develop...feature/add-ical-subscription#diff-5f47db8ad88d6c2319f6c38c2dc1cd112d8a4235e0f857514b816280d1b70fd2R103-R105

Reply:
I think now I got it :)